packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationUtils.kt

```diff
@@ -0,0 +1,126 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.amplify.amplify_push_notifications
+
+import android.content.Intent
+import android.os.Bundle
+import com.amplifyframework.pushnotifications.pinpoint.utils.NotificationPayload
+import com.amplifyframework.pushnotifications.pinpoint.utils.PushNotificationsConstants
+import com.amplifyframework.pushnotifications.pinpoint.utils.toNotificationsPayload
+import com.google.firebase.messaging.RemoteMessage
+
+// TODO: Revisit this file and remove un-used functions
+
+private const val PAYLOAD_KEY = "payload"
+
+fun Intent.isPushNotificationIntent(): Boolean {
+    return this.extras?.containsKey("google.message_id") == true
+}
+fun RemoteMessage.isRemoteMessageSupported(): Boolean {
+    return !this.data[PushNotificationsConstants.PINPOINT_CAMPAIGN_CAMPAIGN_ID].isNullOrEmpty() or
+            !this.data[PushNotificationsConstants.JOURNEY_ID].isNullOrEmpty()
+}
+
+fun RemoteMessage.asPayload(): NotificationPayload {
+    val messageId = this.messageId
+    val senderId = this.senderId
+    val sendTime = this.sentTime
+    val data = this.data
+    val body = this.notification?.body
+        ?: data[PushNotificationsConstants.MESSAGE_ATTRIBUTE_KEY]
+        ?: data[PushNotificationsConstants.PINPOINT_NOTIFICATION_BODY]
+    val title = this.notification?.title
+        ?: data[PushNotificationsConstants.TITLE_ATTRIBUTE_KEY]
+        ?: data[PushNotificationsConstants.PINPOINT_NOTIFICATION_TITLE]
+    val imageUrl = this.notification?.imageUrl?.toString()
+        ?: data[PushNotificationsConstants.IMAGEURL_ATTRIBUTE_KEY]
+        ?: data[PushNotificationsConstants.PINPOINT_NOTIFICATION_IMAGEURL]
+    val action: HashMap<String, String> = HashMap()
+    data[PushNotificationsConstants.PINPOINT_OPENAPP]?.let {
+        action.put(PushNotificationsConstants.PINPOINT_OPENAPP, it)
+    }
+    data[PushNotificationsConstants.PINPOINT_URL]?.let {
+        // force HTTPS URL scheme
+        val urlHttps = it.replaceFirst("http://", "https://")
+        action.put(PushNotificationsConstants.PINPOINT_URL, urlHttps)
+    }
+    data[PushNotificationsConstants.PINPOINT_DEEPLINK]?.let {
+        action.put(PushNotificationsConstants.PINPOINT_DEEPLINK, it)
+    }
+
+    return NotificationPayload {
+        notification(messageId, senderId, sendTime)
+        notificationContent(title, body, imageUrl)
+        notificationOptions(PushNotificationsConstants.DEFAULT_NOTIFICATION_CHANNEL_ID)
+        tapAction(action)
+        silentPush = data[PushNotificationsConstants.PINPOINT_NOTIFICATION_SILENTPUSH].equals("1")
+        rawData = HashMap(data)
+    }
+}
+
+fun getPayloadFromExtras(extras: Bundle?): NotificationPayload? {
```

Comment:
Change to extension method:

```suggestion
val Bundle.notificationPayload: NotificationPayload?
  get() = {
```